wfe2/wfe.go

```diff
@@ -611,6 +615,83 @@ func link(url, relation string) string {
 	return fmt.Sprintf("<%s>;rel=\"%s\"", url, relation)
 }
 
+// checkNewAccountLimits checks whether sufficient limit quota exists for the
+// creation of a new account from the given IP address. If an error is
+// encountered during the check, it is logged but not returned.
+//
+// TODO(#5545): For now we're simply exercising the new rate limiter codepath.
+// This should eventually return a berrors.RateLimit error containing the retry
+// after duration among other information available in the ratelimits.Decision.
+func (wfe *WebFrontEndImpl) checkNewAccountLimits(ctx context.Context, ip net.IP) {
+	if wfe.limiter == nil {
+		// Limiter is disabled.
+		return
+	}
+	decision, err := wfe.limiter.Spend(ctx, ratelimits.NewRegistrationsPerIPAddress, ip.String(), 1)
+	if err != nil {
+		if errors.Is(err, ratelimits.ErrLimitDisabled) {
+			return
+		}
+		wfe.log.Warningf("checking %s rate limit: %s", ratelimits.NewRegistrationsPerIPAddress, err)
```

Comment:
From conversation this morning: this is fine as-is, but add a TODO saying that this log line should be removed when this error is upgraded to fail-closed, and will be logged in the WFE's normal error log line.

Reply:
> We shouldn't log these at warning level. Info at the highest, or not at all since adding one log line for every request is a lot.

From our offline conversation:
This is logging each error which occurs, not each request. This is important because we will returning and logging errors until key-value rate limits are authoritative. Without this line we're still categorizing errors and incrementing them as metrics within the rate limits Redis source itself, but this line allows us to actually see what the error was. Errors being logged here should be exceedingly rare (currently <0.04%).

> From conversation this morning: this is fine as-is, but add a TODO saying that this log line should be removed when this error is upgraded to fail-closed, and will be logged in the WFE's normal error log line.

Agreed.